### PR TITLE
Add protos for verification options

### PR DESCRIPTION
**Pull request by rbehjati**
First step towards a flexible verification library for verifying remote attestation evidence and reports.

### CODE REVIEW

File: oak_remote_attestation_verification/proto/verification.proto [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+// limitations under the License.
+//
+
+syntax = "proto3";
+
+package oak.verification.v1;
+
+enum TeePlatform {
+  UNSPECIFIED = 0;
+  ANY = 1;
```
Comment (conradgrobler) [inline oak_remote_attestation_verification/proto/verification.proto:23]: What is the expected behavior when this option is specified? Perhaps add a comment to clarify.

**Review by conradgrobler — APPROVED** [review]

File: oak_remote_attestation_verification/proto/verification.proto [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+message BinaryVerificationOptions {
+    TransparencyVerificationOptions transparency_verification_options = 1;
+    // Expected TEE platform.
+    TeePlatform tee_platform = 2;
+}
+
+message TransparencyVerificationOptions {
+   oneof option {
+        SkipBinaryTransparencyVerification skip_binary_transparency_verification = 1;
+        BinaryTransparencyVerificationOptions binary_transparency_verification_options = 2;
```
Comment (ipetr0v) [inline oak_remote_attestation_verification/proto/verification.proto:38]: Is it `BinaryVerificationOptions` ?

File: oak_remote_attestation_verification/proto/verification.proto [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    }
+}
+
+// The presence of this option in TransparencyVerificationOptions implies that an endorsement
+// statement must be present, its subject must be as expected, and it must be signed by a trusted
+// entity. Therefore, no additional fields are required for verifying these details.
+message RequireBinaryTransparencyVerification {
```
Comment (ipetr0v) [inline oak_remote_attestation_verification/proto/verification.proto:45]: This message is not used anywhere

**Review by ipetr0v — APPROVED** [review]

**Review by ipetr0v — APPROVED** [review]

**Review by rbehjati — COMMENTED** [review]
Thanks for the reviews. I am some changes, and tried to reflect the DICE layers here too. Please take another look.

**Review by conradgrobler — APPROVED** [review]

**Pull request merged**